internal/service/dynamodb/contributor_insights.go

```diff
@@ -0,0 +1,161 @@
+package dynamodb
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
+)
+
+func ResourceContributorInsights() *schema.Resource {
+	return &schema.Resource{
+		CreateWithoutTimeout: resourceContributorInsightsCreate,
+		ReadWithoutTimeout:   resourceContributorInsightsRead,
+		DeleteWithoutTimeout: resourceContributorInsightsDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"index_name": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Optional: true,
+			},
+			"table_name": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceContributorInsightsCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).DynamoDBConn
+
+	in := &dynamodb.UpdateContributorInsightsInput{
+		ContributorInsightsAction: aws.String(dynamodb.ContributorInsightsActionEnable),
+	}
+
+	if v, ok := d.GetOk("table_name"); ok {
+		in.TableName = aws.String(v.(string))
+	}
+
+	var indexName string
+	if v, ok := d.GetOk("index_name"); ok {
+		in.IndexName = aws.String(v.(string))
+		indexName = v.(string)
+	}
+
+	out, err := conn.UpdateContributorInsightsWithContext(ctx, in)
+	if err != nil {
+		return diag.Errorf("creating dynamodb ContributorInsights for table (%s): %s", d.Get("table_name").(string), err)
```

Comment:
In other DynamoDB resources, e.g. https://github.com/hashicorp/terraform-provider-aws/blob/690cee18d1d3aefce2270cc88cf21e05b6ae6a06/internal/service/dynamodb/table.go#L547 we are using `DynamoDB` in any error messages.

```suggestion
		return diag.Errorf("creating DynamoDB ContributorInsights for table (%s): %s", d.Get("table_name").(string), err)
```

Reply:
Same in other CRUD handler methods.

Reply:
updated to follow convention of other errors